src/strategies/results.js

```diff
@@ -0,0 +1,44 @@
+import F from 'futil'
+import _ from 'lodash/fp'
+import { runWith, flattenProp } from '../utils'
+
+// Will go away once results no longer wraps context in `response`
+let resultField = (field, node) =>
+  F.cascade([`response.${field}`, field], node.context)
+
+export default async ({ service, tree, ...node }) => {
+  let { page = 1, pageSize = 100 } = node
+  let run = props =>
+    runWith(
+      service,
+      tree,
+      F.compactObject({ key: 'results', type: 'results', ...node, ...props })
+    )
+  let scrollId
+  let totalRecords = resultField(
+    'totalRecords',
+    await run({ pageSize: 1, page: 1 })
+  )
+
+  let result = {
+    getTotalRecords() {
+      return totalRecords
+    },
+    async *[Symbol.asyncIterator]() {
+      while (page <= Math.ceil(totalRecords / pageSize)) {
+        let node = await run({ page, scrollId, skipCount: true })
```

Comment:
I just noticed that you are no longer setting `result.node` - this is critical to make it easier to debug